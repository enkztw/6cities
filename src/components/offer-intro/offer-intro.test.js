import React from 'react';
import renderer from 'react-test-renderer';
import Intro from './offer-intro.jsx';

const offer = {
  id: 1,
  city: {
    name: `Amsterdam`,
    location: {
      latitude: 52.370216,
      longitude: 4.895168,
      zoom: 10
    }
  },
  previewImage: `img/apartment-01.jpg`,
  images: [`img/apartment-02.jpg`, `img/apartment-03.jpg`],
  title: `Beautiful & luxurious studio at great location`,
  isFavorite: false,
  isPremium: false,
  rating: 1.8,
  type: `Apartment`,
  bedrooms: 3,
  maxAdults: 4,
  price: 100,
  goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
  host: {
    id: 3,
    isPro: true,
    name: `Angelina`,
    avatarUrl: `img/avatar-angelina.jpg`
  },
  description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
  location: {
    latitude: 52.35514938496378,
    longitude: 4.673877537499948,
    zoom: 8
  }
};

describe(`Render <Intro />`, () => {
  it(`<Intro /> should render`, () => {
    const tree = renderer.create(<Intro {...offer} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`<Intro /> should render with premium and book marks`, () => {
    const tree = renderer.create(<Intro {...offer} isPremium={true} isFavorite={true} />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
