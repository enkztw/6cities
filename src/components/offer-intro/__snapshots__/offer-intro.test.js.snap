// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <Intro /> <Intro /> should render 1`] = `
Array [
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    />
    <button
      className="property__bookmark-button button"
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "100%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      4.51
    </span>
  </div>,
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      Apartment
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      3
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
       adults
    </li>
  </ul>,
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      1337
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>,
]
`;

exports[`Render <Intro /> <Intro /> should render with premium and book marks 1`] = `
Array [
  <div
    className="property__name-wrapper"
  >
    <h1
      className="property__name"
    />
    <button
      className="property__bookmark-button button"
      type="button"
    >
      <svg
        className="property__bookmark-icon"
        height="33"
        width="31"
      >
        <use
          xlinkHref="#icon-bookmark"
        />
      </svg>
      <span
        className="visually-hidden"
      >
        To bookmarks
      </span>
    </button>
  </div>,
  <div
    className="property__rating rating"
  >
    <div
      className="property__stars rating__stars"
    >
      <span
        style={
          Object {
            "width": "100%",
          }
        }
      />
      <span
        className="visually-hidden"
      >
        Rating
      </span>
    </div>
    <span
      className="property__rating-value rating__value"
    >
      4.51
    </span>
  </div>,
  <ul
    className="property__features"
  >
    <li
      className="property__feature property__feature--entire"
    >
      Apartment
    </li>
    <li
      className="property__feature property__feature--bedrooms"
    >
      3
       Bedrooms
    </li>
    <li
      className="property__feature property__feature--adults"
    >
      Max 
       adults
    </li>
  </ul>,
  <div
    className="property__price"
  >
    <b
      className="property__price-value"
    >
      €
      1337
    </b>
    <span
      className="property__price-text"
    >
       night
    </span>
  </div>,
]
`;
